sympy/sets/sets.py

```diff
@@ -216,7 +216,15 @@ def _complement(self, other):
             return S.EmptySet
 
         elif isinstance(other, FiniteSet):
-            return FiniteSet(*[el for el in other if self.contains(el) != True])
+            both_l = non_num, num = [], []
+            for el in other:
+                both_l[el.is_number].append(el)
+            non_num = FiniteSet(*non_num)
+            out_range = FiniteSet(*[el for el in num if self.contains(el) == False])
```

Comment:
`!=True` means "keep it if in doubt" whereas `== False` means "keep it if you are sure". Which do you actually want? A test that discriminates between the two would be good to add.